.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Java CI with Maven and Release Artifact
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
If I understand correctly, this means that only pushes to the `main` branch will lead to an automatic build/test, right? Instead, I would suggest to do this for all branches, because that way build/test failures will be detected as early as possible.